version: '2.4'

services:
  user-profile:
    build:
      context: ../user-profile
      dockerfile: Dockerfile
    restart: on-failure
    networks:
      - user-profile-net
    ports:
      - "5000:5000"
    environment:
      - DATABASE_USERNAME=${DATABASE_USERNAME}
      - DATABASE_PASSWORD=${DATABASE_PASSWORD}
      - DATABASE_DOMAIN=${DATABASE_DOMAIN}
      - DATABASE_SCHEMA=${DATABASE_SCHEMA}
      - DATABASE_PORT=${DATABASE_PORT}
      - FLASK_APP=user_profile_service
      - FLASK_DEBUG=1
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - ../user-profile:/home/service

  db:
    image: postgres:${POSTGRES_VERSION}
    restart: always
    networks:
      - user-profile-net
    ports:
      - "5433:5432"
    environment:
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_DB=${POSTGRES_DB}
    volumes:
      - db-data:/var/lib/postgresql/data
    healthcheck:
      test: /usr/bin/pg_isready
      interval: 5s
      timeout: 10s
      retries: 120
    user: postgres

volumes:
  db-data:
    name: user-profile-data

networks:
  user-profile-net:
    name: user-profile-net
    driver: bridge